### ISSUE TITLE
Nested string interpolation failing

### ISSUE THREAD

**Issue by jeffchan** [decaffeinate/decaffeinate#133]
This is happening on latest master.

Following is failing:
```
check('"#{a("#{b}")}"', '`${a(`${b}`)}`');
```

```
  1) string interpolation handles nested interpolations:
     Error: cannot traverse unknown node type: ConcatOp
    at childPropertyNames (dist/decaffeinate.cjs.js:1595:11)
    at traverse (dist/decaffeinate.cjs.js:1488:47)
    at dist/decaffeinate.cjs.js:1479:11
    at Array.forEach (native)
    at dist/decaffeinate.cjs.js:1477:15
    at Array.forEach (native)
    at descend (dist/decaffeinate.cjs.js:1474:32)
    at traverse (dist/decaffeinate.cjs.js:1491:5)
    at dist/decaffeinate.cjs.js:1479:11
    at Array.forEach (native)
    at dist/decaffeinate.cjs.js:1477:15
    at Array.forEach (native)
    at descend (dist/decaffeinate.cjs.js:1474:32)
    at traverse (dist/decaffeinate.cjs.js:1491:5)
    at dist/decaffeinate.cjs.js:1479:11
    at Array.forEach (native)
    at dist/decaffeinate.cjs.js:1477:15
    at Array.forEach (native)
    at descend (dist/decaffeinate.cjs.js:1474:32)
    at traverse (dist/decaffeinate.cjs.js:1491:5)
    at dist/decaffeinate.cjs.js:1483:9
    at Array.forEach (native)
    at descend (dist/decaffeinate.cjs.js:1474:32)
    at traverse (dist/decaffeinate.cjs.js:1491:5)
    at parse$1 (dist/decaffeinate.cjs.js:5621:3)
    at convert (dist/decaffeinate.cjs.js:5777:13)
    at check (test/support/check.js:19:57)
    at Context.<anonymous> (test/string_interpolation_test.js:51:25)
```

Simply adding `ConcatOp` to `traverse.js` and to `patches/index.js` mapping doesn't work. The JS string produced is still invalid:
```
`${a("#{b}`;)}"
```

This could be an issue with decaffeinate-parser, so I wanted to check first before attempting a fix.

**Comment by eventualbuddha** [comment]
What I usually do to debug potential parser bugs is to go to my checked-out version of decaffeinate-parser and feed the CoffeeScript code to `./script/print-ast.js`. I had changed `ConcatOp` to `TemplateLiteral`, but I guess I didn't handle nested ones properly. If you feel like taking a stab at that in decaffeinate-parser that'd be great! The first step is just adding your example as a test until `test/examples`. I created https://github.com/decaffeinate/decaffeinate-parser/issues/3 to track the issue.
